archive/utils_archive/filter_activities.py

```diff
@@ -0,0 +1,48 @@
+from utils.Recording import Recording
+import pandas as pd
+
+
+def filter_short_activities(recordings: 'list[Recording]', threshhold: int = 3, strategy: int = 0) -> 'list[Recording]':
+    """
+    Replaces activities shorter than threshhold by [value]. [value] depends on strategy.
+    strategy 0: replaces short activities with previous activity
+    strategy 1: replaces short activities with 'null-activity'
+
+    threshold: number of seconds
+    """
+
+    if strategy != 0 and strategy != 1:
+        raise ValueError('strategy has to be 0 or 1')
+    
+    for recording in recordings:
+        indizes = []
+        length = recording.activities.shape[0]
+
+        for i in range(length-1):
+            if recording.activities.iloc[i] != recording.activities.iloc[i+1]:
+                indizes.append(i+1)
```

Comment:
Instead of looping over all activities, it is way quicker with numpy:

```python
activities = recording.activities
indizes = np.where(activities[:-1] != activities[1:])[0] + 1
```
This compares all elements with their neighbour and returns the indices where they differ. To get the first different element, we add 1

Also see this SO answer: https://stackoverflow.com/a/19125898